src/account_manager/operations/client.rs

```diff
@@ -0,0 +1,168 @@
+
+        let mut accounts = self.accounts.write().await;
+        for account in accounts.iter_mut() {
+            account.update_account_with_new_client(new_client.clone()).await?;
+        }
+
+        #[cfg(feature = "storage")]
+        {
+            // Update account manager data with new client options
+            let account_manager_builder = AccountManagerBuilder::from_account_manager(self)
+                .await
+                .with_client_options(options);
+
+            self.storage_manager
+                .lock()
+                .await
+                .save_account_manager_data(&account_manager_builder)
+                .await?;
+        }
+        Ok(())
+    }
+
+    /// Get the used client options.
+    pub async fn get_client_options(&self) -> ClientOptions {
+        self.client_options.read().await.clone()
+    }
+
+    /// Get the node info.
+    pub async fn get_node_info(&self) -> crate::Result<NodeInfoWrapper> {
+        let accounts = self.accounts.read().await;
+
+        // Try to get the Client from the first account and only build the Client if we have no account
+        let node_info_wrapper = match &accounts.first() {
+            Some(account) => account.client.get_info().await?,
+            None => self.client_options.read().await.clone().finish()?.get_info().await?,
+        };
+
+        Ok(node_info_wrapper)
+    }
+
+    /// Update the authentication for a node.
+    pub async fn update_node_auth(&self, url: Url, auth: Option<NodeAuth>) -> crate::Result<()> {
+        log::debug!("[update_node_auth]");
+        let mut client_options = self.client_options.write().await;
+
+        if let Some(primary_node) = &client_options.node_manager_builder.primary_node {
+            let (node_url, disabled) = match &primary_node {
+                NodeDto::Url(node_url) => (node_url, false),
+                NodeDto::Node(node) => (&node.url, node.disabled),
+            };
+
+            if node_url == &url {
+                client_options.node_manager_builder.primary_node = Some(NodeDto::Node(Node {
+                    url: url.clone(),
+                    auth: auth.clone(),
+                    disabled,
+                }));
+            }
+        }
+
+        if let Some(primary_pow_node) = &client_options.node_manager_builder.primary_pow_node {
+            let (node_url, disabled) = match &primary_pow_node {
+                NodeDto::Url(node_url) => (node_url, false),
+                NodeDto::Node(node) => (&node.url, node.disabled),
+            };
+
+            if node_url == &url {
+                client_options.node_manager_builder.primary_pow_node = Some(NodeDto::Node(Node {
+                    url: url.clone(),
+                    auth: auth.clone(),
+                    disabled,
+                }));
+            }
+        }
+
+        if let Some(permanodes) = &client_options.node_manager_builder.permanodes {
+            let mut new_permanodes = HashSet::new();
```

Comment:
Can't we just edit them instead of creating a new set ?

Reply:
Is that possible? Didn't find a way https://stackoverflow.com/a/35970285

Reply:
Ah yeah, didn't think about that but it makes sense